CMakeLists.txt

```diff
@@ -67,36 +89,126 @@ else(APPLE)
     set(CMAKE_DISABLE_PRECOMPILE_HEADERS ON)
 endif()
 
+#-------------------------------------------------------------------
+# Setup CMake Includes
+#-------------------------------------------------------------------
+
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
 include(AddStaticResource)
 include(AddShader)
 include(AddStaticResource)
-include(Versioning)
+include(ShowBuildTargetProperties)
+if(WIN32)
+    include(GetWindowsNtVersion)
+endif()
+
+#-------------------------------------------------------------------
+# Setup Dependency Manager: VCPKG
+#-------------------------------------------------------------------
+
+include(SetupVcpkg)
+
+#-------------------------------------------------------------------
+# Find Dependencies
+#-------------------------------------------------------------------
+
+#
+# RenderDoc - non-vcpkg, directly build from externals
+#
+add_subdirectory(external/RenderDoc)
+
+#
+# date
+#
+find_package(date CONFIG REQUIRED)
+
+#
+# fmt
+#
+find_package(fmt CONFIG REQUIRED)
+
+#
+# GoogleTest
+#
+#find_package(GTest)
+
+#
+# GoogleTest - non-vcpkg, directly build from externals
+#
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+set(gtest_force_shared_crt ON)
+
+# Don't install GTest libs when we run `install`.
+set(INSTALL_GTEST OFF CACHE BOOL "Enable installation of googletest." FORCE)
+
+add_subdirectory(external/googletest)
+
+#
+# Vulkan Memory Allocator
+#
+find_package(unofficial-vulkan-memory-allocator CONFIG REQUIRED)
+
+#
+# Vulkan SDK Headers
+#
+if(NOT DEFINED ENV{VULKAN_SDK})
+    message(FATAL_ERROR
+        "Please set the VULKAN_SDK environment variable to the directory in which the SDK is installed.\n"
+        "It looks like C:\VulkanSDK\1.0.0.0, but with the version number being the version that was installed.\n"
+        "set VULKAN_SDK=\"C:\\VulkanSDK\\1.0.0.0\"\n")
+endif()
+find_package(Vulkan REQUIRED)
+
+#-------------------------------------------------------------------
+# Setup Build Targets
+#-------------------------------------------------------------------
+
+add_library(ttauri)
+add_library(ttauri::tauri ALIAS ttauri)
```

Comment:
What is this about?

Reply:
This sets up a CMake namespace for the library. This enables the user of your library to use namespaced imported targets.
Which is the [recommended way](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html?highlight=namespaced) to export libraries.

- Basically to give the target a more structured name and stop to care about the read header file names.
- It also allows to group things under the namespace, e.g. one could introduce `ttauri::docs` or `ttauri::subcomponent`. 
For instance, the `date::date` library has a timezone subcomponent `date::date-tz`. 
- The real advantage is that you get better CMake debug output/diagnostics, when working with your library in another project via `target_link_libraries`.